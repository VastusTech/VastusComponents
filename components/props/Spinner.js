import React from "react";
import {Message, Icon} from "semantic-ui-react";

type Props = {
    loading: boolean
}

const Spinner = (props: Props) => {
    if (props.loading) {
        return (
            <Message icon>
                <Icon name='spinner' size="small" loading />
                <Message.Content>
                    <Message.Header>
                        Loading...
                    </Message.Header>
                </Message.Content>
            </Message>
        );
    }
    return null;
};

Spinner.defaultProps = {
    loading: true
};

export default Spinner;
