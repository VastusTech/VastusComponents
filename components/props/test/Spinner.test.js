import React from 'react';
import { shallow } from 'enzyme';
import TestConfig, {store} from "../../../../TestConfig";
import Spinner from "../Spinner";

TestConfig();
it('renders without crashing', () => {
    const component = shallow(<Spinner loading={true} store={store()}/>);
    expect(component).toMatchSnapshot();
});
