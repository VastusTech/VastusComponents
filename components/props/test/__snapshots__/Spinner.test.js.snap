// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    loading={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Icon
          as="i"
          loading={true}
          name="spinner"
          size="small"
        />,
        <MessageContent>
          <MessageHeader>
            Loading...
          </MessageHeader>
        </MessageContent>,
      ],
      "icon": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "loading": true,
          "name": "spinner",
          "size": "small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <MessageHeader>
            Loading...
          </MessageHeader>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Icon
            as="i"
            loading={true}
            name="spinner"
            size="small"
          />,
          <MessageContent>
            <MessageHeader>
              Loading...
            </MessageHeader>
          </MessageContent>,
        ],
        "icon": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "loading": true,
            "name": "spinner",
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <MessageHeader>
              Loading...
            </MessageHeader>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Loading...",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
